### PR TITLE
WIP: enable user to configure netpermission while create w cluster

### PR DESCRIPTION
**Pull request by nikhilbarge**
### What this PR does / why we need it
This pr will enable tkgm user to configure netpermission while create wl clusters
### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
This pr will enable tkgm user to confgure csi net-permissions 
```

### CODE REVIEW

**Comment by alfredthenarwhal** [comment]
> @nikhilbarge:  /test install-vc7 --graph tkg-v2.2.0-zshippable.yaml
Commit: ebce04f

Tests passed! Build no: 4209

File: providers/config_default.yaml [hunk 1]
```diff
@@ -155,6 +155,18 @@ VSPHERE_AZ_0:
...
+#! example:
+#! NETPERMISSIONS:
+#!   A:
+#!     ips: "*"
+#!     permissions: READ_WRITE
+#!     rootsquash: false
+NETPERMISSIONS:
+
+#! The namespace csi components are to be deployed in
+CSINAMESPACE: ""
```
Comment (jeffwubj) [inline providers/config_default.yaml:168]: nit: CSI_NAMESPACE

Comment (nikhilbarge) [reply providers/config_default.yaml]: changed to CSI_NAMESPACE

**Review by jeffwubj — APPROVED** [review]

**Comment by wjun** [comment]
What integration tests have been done for this pr?

**Comment by nikhilbarge** [comment]
to test this i have run pipeline  install-vc7  also manually tested

File: providers/config_default.yaml [hunk 2]
```diff
@@ -155,6 +155,18 @@ VSPHERE_AZ_0:
...
 VSPHERE_AZ_2:
 
+#! Settings to provide netpermissions for vpshere-csi file volumes
+#! example:
+#! NETPERMISSIONS:
+#!   A:
+#!     ips: "*"
+#!     permissions: READ_WRITE
+#!     rootsquash: false
+NETPERMISSIONS:
```
Comment (DanielXiao) [inline providers/config_default.yaml:165]: A map object does not work here and you can expose 3 variables.

File: providers/config_default.yaml [hunk 3]
```diff
@@ -155,6 +155,18 @@ VSPHERE_AZ_0:
...
+
+#! The namespace csi components are to be deployed in
+CSI_NAMESPACE: ""
```
Comment (DanielXiao) [inline providers/config_default.yaml:168]: CSI namespace is controlled by addon manager and should not expose to users

**Review by DanielXiao — CHANGES_REQUESTED** [review]
Please split the map variable into multiple variables.

File: providers/config_default.yaml (hunk 2 above)
Comment (nikhilbarge) [reply providers/config_default.yaml:165]: will update as per suggestion

File: providers/config_default.yaml (hunk 3 above)
Comment (nikhilbarge) [reply providers/config_default.yaml:168]: will remove

File: providers/config_default.yaml (hunk 2 above)
Comment (wjun) [reply providers/config_default.yaml:165]: @DanielXiao @nikhilbarge this is a typical case that the addon config will include a map struct. I do not think it should be exposed to config file with 3 variables. Instread, workload cluster users should create a new clusterbootstrap to include this optional netpermissions when creating a workload cluster, or edit the vpsherecsiconfig to include this optional netpermissions after creating a management cluster or a workload cluster. + @shyaamsn @adduarte @12345lcr

Comment (DanielXiao) [reply providers/config_default.yaml:165]: Got it, it is a map[string]*NetPermissionConfigs type. We can also use a legacy style variable NETPERMISSIONS_BASE64 to encode the yaml object then decode it to the field netPermissions. It is also acceptable if it is not supported through legacy configuration file.

**Comment by alfredthenarwhal** [comment]
> @nikhilbarge:  /test install-vc7 --graph tkg-v2.2.0-zshippable.yaml
Commit: 08674dc

Tests failed! Build no: 4228

**Review by DanielXiao — APPROVED** [review]

**Review by DanielXiao — CHANGES_REQUESTED** [review]
Could you add format validation code in tanzu cli on NETPERMISSIONS_BASE64 to ensure it can be converted to map[string]*NetPermissionConfig? Please also build tanzu cli and manually test the following scenarios:
1. Input configuration file does not contain NETPERMISSIONS_BASE64.
2. NETPERMISSIONS_BASE64 is a valid format.
3. NETPERMISSIONS_BASE64 is not a valid format.

File: providers/config_default.yaml [hunk 4]
```diff
@@ -155,6 +155,9 @@ VSPHERE_AZ_0:
 VSPHERE_AZ_1:
 VSPHERE_AZ_2:
 
+#! Settings to provide netpermissions for vpshere-csi file volumes as a base64-encoded string
+NETPERMISSIONS_BASE64:
```
Comment (DanielXiao) [inline providers/config_default.yaml:159]: The default value is none rather than empty string.

**Comment by wjun** [comment]
@nikhilbarge If you have tested the clusterbootstrap approach, please drop this pr as cli config file is not a place for optional addon config parameters.

**Review by wjun — CHANGES_REQUESTED** [review]
@nikhilbarge If you have tested the clusterbootstrap approach, please drop this pr as cli config file is not a place for optional addon config parameters.

**Comment by nikhilbarge** [comment]
> @nikhilbarge If you have tested the clusterbootstrap approach, please drop this pr as cli config file is not a place for optional addon config parameters.
ok will close this pr

**Comment by alfredthenarwhal** [comment]
Created cherry-pick PR

**Pull request closed without merging**